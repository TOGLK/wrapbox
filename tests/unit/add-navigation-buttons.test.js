const addNavButtons = require("../../app/scripts/add-navigation-buttons");
const EventEmitter = require("events");

const testEmitter = new EventEmitter();

function createTestDocument() {
  for (let i = 0; i < 2; i++) {
    const div = document.createElement("div");
    document.querySelector("body").appendChild(div);
    div.setAttribute("id", `div${i}`);
  }
}

beforeAll(() => {
  createTestDocument();
});

describe("AddNavButtons function", () => {
  test("creates navbuttons section only in defined element", () => {
    addNavButtons("#div0", testEmitter, 0);
    expect(document.querySelector("#nbdiv0").parentNode.id).toBe("div0");
    expect(document.querySelector("#div1").childElementCount).toBe(0);
  });

  test("navbuttons are hidden by default", () => {
    expect(document.querySelector("#nbdiv0").classList.contains("invisible")).toBeTruthy();
  });

  test("navbuttons are visible if parent node is 'active'", () => {
    expect(document.querySelector("#nbdiv0").classList.contains("invisible")).toBeTruthy();
  });

  test("adds 4 buttons to navbuttons element", () => {
    const buttons = document.querySelector("#nbdiv0").childNodes;
    expect(buttons.length).toBe(4);
    for (let i = 0; i < buttons.length; i++) {
      expect(buttons[i]).toMatchSnapshot();
    }
  });
});
