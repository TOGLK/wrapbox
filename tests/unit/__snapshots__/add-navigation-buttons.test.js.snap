// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNavButtons function adds 4 buttons to navbuttons element 1`] = `
<button
  class="mdl-shadow--8dp mdl-button mdl-js-button mdl-color--blue-grey-700 mdl-color-text--white mdl-button--fab mdl-js-ripple-effect"
>
  <i
    class="material-icons"
  >
    navigate_before
  </i>
</button>
`;

exports[`AddNavButtons function adds 4 buttons to navbuttons element 2`] = `
<button
  class="mdl-shadow--8dp mdl-button mdl-js-button mdl-color--blue-grey-700 mdl-color-text--white mdl-button--fab mdl-js-ripple-effect"
>
  <i
    class="material-icons"
  >
    home
  </i>
</button>
`;

exports[`AddNavButtons function adds 4 buttons to navbuttons element 3`] = `
<button
  class="mdl-shadow--8dp mdl-button mdl-js-button mdl-color--blue-grey-700 mdl-color-text--white mdl-button--fab mdl-js-ripple-effect"
>
  <i
    class="material-icons"
  >
    refresh
  </i>
</button>
`;

exports[`AddNavButtons function adds 4 buttons to navbuttons element 4`] = `
<button
  class="mdl-shadow--8dp mdl-button mdl-js-button mdl-color--blue-grey-700 mdl-color-text--white mdl-button--fab mdl-js-ripple-effect"
>
  <i
    class="material-icons"
  >
    navigate_next
  </i>
</button>
`;
